pkg/didmethod/httpbinding/resolver.go

```diff
@@ -0,0 +1,139 @@
+	errors "golang.org/x/xerrors"
+)
+
+var logger = log.New("aries-framework/didmethod/httpbinding")
+
+// HTTPClientOpts holds options for the HTTP transport implementation of DIDResolver
+// it has a required http.Client instance and optional basePath which is appended to the remove DID Resolver URL
+type HTTPClientOpts struct {
+	client   *http.Client
+	basePath string
+}
+
+// HTTPOpt is an outbound HTTP transport option
+type HTTPOpt func(opts *HTTPClientOpts)
+
+// WithHTTPClientInstance option is for creating an Outbound HTTP transport using an http.Client instance
+func WithHTTPClientInstance(client *http.Client) HTTPOpt {
+	return func(opts *HTTPClientOpts) {
+		opts.client = client
+	}
+}
+
+// WithHTTPClientTimeout option is for creating an Outbound HTTP transport using a client timeout value
+func WithHTTPClientTimeout(timeout time.Duration) HTTPOpt {
+	return func(opts *HTTPClientOpts) {
+		opts.client.Timeout = timeout
+	}
+}
+
+// WithHTTPClientTLSConfig option is for creating an Outbound HTTP transport using a tls.Config instance
+func WithHTTPClientTLSConfig(tlsConfig *tls.Config) HTTPOpt {
+	return func(opts *HTTPClientOpts) {
+		opts.client.Transport = &http.Transport{
+			TLSClientConfig: tlsConfig,
+		}
+	}
+}
+
+// WithHTTPClientBasePath options is for setting optional base path, for example "document"
+func WithHTTPClientBasePath(basePath string) HTTPOpt {
+	return func(opts *HTTPClientOpts) {
+		opts.basePath = basePath
+	}
+}
+
+// resolveDID makes DID resolution via HTTP
+func (res *DIDResolver) resolveDID(url string) ([]byte, error) {
+	resp, err := res.client.Get(url)
+	if err != nil {
+		return nil, errors.Errorf("HTTP Get request failed: %w", err)
+	}
+
+	defer func() {
+		e := resp.Body.Close()
+		if e != nil {
+			logger.Errorf("Failed to close response body: %v", e)
+		}
+	}()
+
+	// handle response
+	// TODO support for service endpoint URL resolution
+	if containsDIDDocument(resp) {
+		var gotBody []byte
+		gotBody, err = ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return nil, errors.Errorf("Failed to read response body: %w", err)
+		}
+
+		return gotBody, nil
+
+	} else if notExistentDID(resp) {
+		return nil, errors.Errorf("Input DID does not exist: %w", err)
+	}
+
+	return nil, errors.Errorf("Unsupported response from DID Resolver with status code: %v", resp.StatusCode)
+}
+
+// notExistentDID checks if requested DID is not found on remote DID resolver
+func notExistentDID(resp *http.Response) bool {
+	return resp.StatusCode == http.StatusNotFound
+}
+
+// containsDIDDocument checks weather reply from remote DID resolver contains DID document
+func containsDIDDocument(resp *http.Response) bool {
+	return resp.StatusCode == http.StatusOK && resp.Header.Get("Content-type") == "application/did+ld+json"
+}
+
+// New creates new DIDResolver
+func New(host string, opts ...HTTPOpt) (*DIDResolver, error) {
+	// Apply options
+	clOpts := &HTTPClientOpts{client: http.DefaultClient}
```

Comment:
Let's use `&http.Client{}` instead of `http.DefaultClient`. Any code can mutate the object assigned to `http.DefaultClient` and thus we should not trust this variable.

Reply:
Agree!